ambry-router/src/test/java/com/github/ambry/router/NonBlockingRouterTest.java

```diff
@@ -3136,6 +3136,55 @@ public void testBlobDelete503NoRetrySinceNoSuccessReplica() throws Exception {
     }
   }
 
+  /**
+   * Test Deletion first hits Not_FOUND. Then run ReplicateBlob.
+   * If the PutBlob is compacted already, ReplicateBlob may fail with ID_Deleted.
+   * @throws Exception
+   */
+  @Test
+  public void testReplicateBlobOnDeleteButPutBlobCompacted() throws Exception {
+    String serviceID = "delete-service";
+    MockServerLayout layout = new MockServerLayout(mockClusterMap);
+    String localDcName = "DC3";
+    setRouter(getNonBlockingRouterProperties(localDcName), layout, new LoggingNotificationSystem());
+    setOperationParams();
+    String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel, new PutBlobOptionsBuilder().build())
+        .get(AWAIT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+    // two local replicas are unavailable. Remote replica doesn't have the Blob.
+    DataNodeId sourceDataNode = null;
+    List<ServerErrorCode> serverErrors = new ArrayList<>();
+    serverErrors.add(ServerErrorCode.Blob_Not_Found); // return NOT_FOUND for the first Delete Request
```

Comment:
so does the delete op fail with not_found or ambry_unavlb error ?

Reply:
At server level, the single delete request fails with NOT_FOUND on the remote replica.
At frontend level, router checks all the responses from both the local and remote replica, the final status for deletion is  unavailable.